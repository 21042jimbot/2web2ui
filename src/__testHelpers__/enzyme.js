export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
