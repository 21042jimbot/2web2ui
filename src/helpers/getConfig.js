import _ from 'lodash';
import config from 'src/config';

// This is much easier to stub than using the object directly
const getConfig = path => _.get(config, path);

export default getConfig;
