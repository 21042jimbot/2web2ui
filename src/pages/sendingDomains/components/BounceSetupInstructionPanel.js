import React from 'react';
import ConditionSwitch, { Case, defaultCase } from 'src/components/auth/ConditionSwitch';
import LabelledValue from 'src/components/labelledValue/LabelledValue';
import { hasAccountOptionEnabled } from 'src/helpers/conditions/account';
import getConfig from 'src/helpers/getConfig';
import { resolveReadyFor } from 'src/helpers/domains';
import SetupInstructionPanel from './SetupInstructionPanel';

const BounceSetupInstructionPanel = ({ domain, hasAutoVerifyEnabled, onVerify, verifyCnameLoading }) => {
  const readyFor = resolveReadyFor(domain.status);
  const bounceDomainsConfig = getConfig('bounceDomains');

  return (
    <ConditionSwitch>
      <Case condition={hasAccountOptionEnabled('byoip_customer')}>
        <p>
          placeholder for new instructions
        </p>
      </Case>
      <Case condition={defaultCase}>
        <SetupInstructionPanel
          isAutoVerified={hasAutoVerifyEnabled}
          isVerified={readyFor.bounce}
          isVerifying={verifyCnameLoading}
          onVerify={onVerify}
          recordType="CNAME"
          verifyButtonIdentifier="verify-cname"
        >
          <LabelledValue label="Type">
            <p>
              CNAME
            </p>
          </LabelledValue>
          <LabelledValue label="Hostname">
            <p>
              {domain.id}
            </p>
          </LabelledValue>
          <LabelledValue label="Value">
            <p>
              {bounceDomainsConfig.cnameValue}
            </p>
          </LabelledValue>
        </SetupInstructionPanel>
      </Case>
    </ConditionSwitch>
  );
};

export default BounceSetupInstructionPanel;
