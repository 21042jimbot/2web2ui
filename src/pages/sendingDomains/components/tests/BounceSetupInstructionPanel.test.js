import React from 'react';
import { shallow } from 'enzyme';
import getConfig from 'src/helpers/getConfig';
import BounceSetupInstructionPanel from '../BounceSetupInstructionPanel';

jest.mock('src/helpers/getConfig');

describe('BounceSetupInstructionPanel', () => {
  const subject = (props = {}) => shallow(
    <BounceSetupInstructionPanel
      domain={{
        id: 'example.com',
        status: {}
      }}
      hasAutoVerifyEnabled={false}
      id="example.com"
      onVerify={() => {}}
      verifyCnameLoading={false}
      {...props}
    />
  );

  beforeEach(() => {
    getConfig.mockImplementation(() => ({ cnameValue: 'spark.mail' }));
  });

  it('renders cname setup instructions', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('calls onVerify handler', () => {
    const onVerify = jest.fn();
    const wrapper = subject({ onVerify });

    wrapper
      .find('SetupInstructionPanel[recordType="CNAME"]')
      .simulate('verify');

    expect(onVerify).toHaveBeenCalled();
  });
});
