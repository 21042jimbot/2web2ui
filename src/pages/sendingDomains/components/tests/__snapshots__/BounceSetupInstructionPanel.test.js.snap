// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BounceSetupInstructionPanel renders cname setup instructions 1`] = `
<SetupInstructionPanel
  isAutoVerified={false}
  isVerified={false}
  isVerifying={false}
  onVerify={[Function]}
  recordType="CNAME"
  verifyButtonIdentifier="verify-cname"
>
  <LabelledValue
    bold={false}
    label="Type"
    value="CNAME"
  />
  <LabelledValue
    bold={false}
    label="Hostname"
    value="example.com"
  />
  <LabelledValue
    bold={false}
    label="Value"
    value="spark.mail"
  />
</SetupInstructionPanel>
`;
